monkey/infection_monkey/exploit/island_api_agent_otp_provider.py

```diff
@@ -8,4 +10,10 @@ def __init__(self, island_api_client: IIslandAPIClient):
         self._island_api_client = island_api_client
 
     def get_otp(self) -> str:
-        return self._island_api_client.get_otp()
+        for _ in range(6):
+            try:
+                return self._island_api_client.get_otp()
+            except IslandAPIRequestLimitExceededError:
+                sleep(0.5)
+                continue
+        raise IslandAPIRequestLimitExceededError("Failed to get OTP: Too many requests.")
```

Comment:
Should this, perhaps, be a `RuntimeError` instead? `IAgentOTPProvider` isn't coupled in any way to `IIslandAPIClient`, nor should it be. The docstring`IAgentOTPProvider` will need to be updated with information about what exceptions can be raised.

```suggestion
            try:
                return self._island_api_client.get_otp()
            except IslandAPIRequestLimitExceededError:
                sleep(0.5)
                continue
            except Exception as err:
                raise RuntimeError(err)
        raise RuntimeError("Failed to get OTP: Too many requests.")
```

Reply:
I'm fine with `RuntimeError`. Should we also capture the other errors that could be raised by `get_otp()`?